ryft-ui/src/main/java/com/ryftpay/android/ui/component/RyftCheckBox.kt

```diff
@@ -0,0 +1,49 @@
+package com.ryftpay.android.ui.component
+
+import android.content.Context
+import android.util.AttributeSet
+import android.widget.CheckBox
+import android.widget.LinearLayout
+import android.widget.TextView
+import androidx.appcompat.content.res.AppCompatResources
+import com.ryftpay.android.ui.extension.setOnSingleClickListener
+import com.ryftpay.android.ui.listener.RyftCheckBoxListener
+import com.ryftpay.ui.R
+
+internal class RyftCheckBox @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null
+) : LinearLayout(context, attrs) {
+
+    private lateinit var checkBox: CheckBox
+    private lateinit var label: TextView
+
+    internal fun initialise(
+        text: String,
+        listener: RyftCheckBoxListener
+    ) {
+        checkBox = findViewById(R.id.chk_ryft_internal)
+        checkBox.setOnCheckedChangeListener { _, checked ->
+            handleChecked(checked, listener)
+        }
+
+        label = findViewById(R.id.text_ryft_check_box)
+        label.text = text
+        label.setOnSingleClickListener {
+            checkBox.toggle()
```

Comment:
is there no way to check the checkbox and trigger the callback, e.g. `checkbox.setChecked(checkbox.toggle())`?
would reduce the next 2 lines being repeated

Reply:
good spot, I've simplified this now